src/main.rs

```diff
@@ -195,3 +136,44 @@ fn select_new_runner() -> Option<Runner> {
 
     chosen_runner
 }
+
+fn main() {
+    let cli = Cli::parse();
+
+    let mut cache = RunnerCache::load();
+
+    let chosen = if cli.force_choose_new {
+        let runner = select_new_runner(cli.config_path);
+        match cache {
+            Some(mut cache) => {
+                if runner.is_some() {
+                    cache.add_runner(&runner.as_ref().unwrap());
+                }
+            },
+            None => {
+                eprintln!("Couldn't parse cache, intentionally not overwriting, check it for errors.");
+                std::process::exit(1);
```

Comment:
This is not a behavior I want. It should still be possible to select a runner, without writing the cache if it is corrupted. This is what the normal (no force choose new) version does.